openforcefield/tests/test_chemicalenvironment.py

```diff
@@ -21,198 +28,68 @@ def test_createEnvironments(self):
-        if not OEParseSmarts(qmol, smirks):
-            raise Exception(f"could not parse created SMIRKS {smirks}")
-
-        # Try removing atoms
-        # if it was labeled:
-        removed = torsion.removeAtom(atom1)
-        self.assertFalse(removed, f"removed labeled atom ({atom1.asSMIRKS()}) in {torsion.asSMIRKS()}")
-        removed = torsion.removeAtom(atom3alpha1)
-        self.assertFalse(removed, f"removed bridging atom ({atom3alpha1.asSMIRKS()}) in {torsion.asSMIRKS()}")
-        removed = torsion.removeAtom(atom3beta1)
-        self.assertTrue(removed, f"failed to remove atom ({atom3beta1.asSMIRKS()}) in {torsion.asSMIRKS()} which is removeable")
-
-    def test_parseSMIRKS(self):
+    @pytest.mark.parametrize(['smirks', 'expected_valence', 'expected_chemenv_class'],
+                             [["[#6](-[#1])-[#8]", None, ChemicalEnvironment],
+                              ["[#6&X4&H0:1](-[#1])-[#6&X4]", 'Atom', AtomChemicalEnvironment],
+                              ["[#6&X4&H0:1](-[#1])-[#6&X4:2]", 'Bond', BondChemicalEnvironment],
+                              ["[*:1]-[*:2](-[#6&X4])-[*:3]", 'Angle', AngleChemicalEnvironment],
+                              ["[#6&X4&H0:1](-[#1])-[#6&X4:2]-[#6&X4&H0:3](-[#1])-[#6&X4:4]", 'ProperTorsion',
+                               TorsionChemicalEnvironment],
+                              ["[#1:1]-[#6&X4:2](-[#8:3])-[#1:4]", 'ImproperTorsion', ImproperChemicalEnvironment],
+                              ["[#1:1]-[#6&X4:2](-[#8:3])-[*:4](-[#6&H1])-[#8:5]", None, ChemicalEnvironment],
+                              ["[#6$(*~[#6]=[#8])$(*-,=[#7!-1,#8,#16,#7])]", None, ChemicalEnvironment],
+                              ["CCC", None, ChemicalEnvironment],
+                              ["[#6:1]1(-;!@[#1,#6])=;@[#6]-;@[#6]1", 'Atom', ChemicalEnvironment],
+                              ["C(O-[#7,#8])CC=[*]", None, ChemicalEnvironment],
+                              ["[#6$([#6X4](~[#7!-1,#8!-1,#16!-1,#9,#17,#35,#53])(~[#8]~[#1])):1]-[#6X2H2;+0:2]-,=,:;!@;!#[#7!-1,#8,#16:3]-[#4:4]",
+                               'ProperTorsion', TorsionChemicalEnvironment],
+                              ["[#6$([#6X4](~[#7!-1,#8!-1,#16!-1,#9,#17,#35,#53])(~[#8]~[#1])):1]1=CCCC1", 'Atom', AtomChemicalEnvironment],
+                              ["[*:1]-[#7X3:2](-[#6a$(*1ccc(-[#8-1X1])cc1):3])-[*:4]", 'ImproperTorsion',
+                               ImproperChemicalEnvironment],
+                              ["[#6X4:1]1~[*:2]~[*$(*~[#1]):3]1", 'Angle', AngleChemicalEnvironment],
+                              ["[$([#7]1~[#6]-CC1)]", None, ChemicalEnvironment],
+                              ["[$(c1ccccc1)]", None, ChemicalEnvironment],
+                              # The next two tests are for ring-closing bonds
+                              ["[H][C@:4]1(C(C([C:3]([N:2]1[C:1](=O)C([H])([H])[H])([H])[H])([H])[H])([H])[H])C=O",
+                               'ImproperTorsion', ChemicalEnvironment],
```

Comment:
Nice close reading. In this case, we want to test that "a smirks which could satisfy a subclass of `ChemicalEnvironment` also satisfies the base class". It took me a little bit to wrap my head around it too. I've added a comment to the test case to avoid future confusion.